### PR TITLE
Fix post pagination

### PR DESCRIPTION
**Pull request by smyzin**
- moved `suspense` inside the component
- added additional suspense to Replies
- fixed pagination

### CODE REVIEW

File: src/components/BlockchainList/PostsList.tsx [hunk 1]
```diff
@@ -33,20 +33,17 @@ export function PostListSuspended() {
...
-
   const postsLength = posts.length
+  const { paginated, skip, setSkip } = usePaginated(posts, limit)
 
   if (!thread) return <LoadingList text="Fetching thread posts..." />
-  if (postsLength === 0) return <NoPosts />
   if (hashId) useScrollToAnchor({ callback: flashingThread })
 
-  return (
+  return paginated.length > 0 ? (
```
Comment (T-Damer) [inline src/components/BlockchainList/PostsList.tsx:42]: Let's extract `paginated.length` into `const`

File: src/components/BlockchainList/ThreadPart.tsx [hunk 2]
```diff
@@ -110,3 +139,25 @@ export default function ({
...
+  owner,
+  canReply,
+}: {
+  threadId: number
+  postId: number
+  limitThread?: number
+  owner: string
+  canReply?: boolean
+}) {
+  const props = { threadId, postId, limitThread, owner, canReply }
```
Comment (T-Damer) [inline src/components/BlockchainList/ThreadPart.tsx:156]: Good one

**Review by MixailE — APPROVED** [review]

File: src/components/BlockchainList/ThreadPart.tsx [hunk 3]
```diff
@@ -20,101 +24,92 @@ const wrapper = classNamesToString(
...
+function ThreadPartSuspended({
   threadId,
-  replyingTo,
   postId,
   limitThread,
-  threadMerkleRoot,
+  owner,
   canReply,
 }: {
   threadId: number
```
Comment (T-Damer) [inline src/components/BlockchainList/ThreadPart.tsx:34]: Should we extract those props into `interface`?

File: src/components/BlockchainList/ThreadPart.tsx [hunk 4]
```diff
@@ -20,101 +24,92 @@ const wrapper = classNamesToString(
...
 }
 
-export default function ({
-  threadId,
-  postId,
-  limitThread,
-  owner,
-  canReply,
-}: {
+export default memo<{
```
Comment (T-Damer) [inline src/components/BlockchainList/ThreadPart.tsx:96]: Do we still need memoization?

File: src/stores/FarcasterStore.ts [hunk 5]
```diff
@@ -6,29 +6,35 @@ import postIdsStatuses from 'stores/PostIdsStatuses'
...
 
-export function fetchFarcasterThread(postId: number) {
-  if (typeof farcasterStore.threads[postId] !== 'undefined') return
-  const request = fetchThreadByPostId(postId)
-  farcasterStore.threads[postId] = request.then(({ casts }) =>
-    casts.map((cast) => cast.merkleRoot)
-  )
-  void request.then(({ casts, merkleRoot }) => {
+export async function fetchFarcasterThread(postId: number) {
+  if (farcasterStore.requested[postId]) return
```
Comment (T-Damer) [inline src/stores/FarcasterStore.ts:17]: Maybe let's extract this into `const`?

File: src/stores/FarcasterStore.ts [hunk 6]
```diff
@@ -6,29 +6,35 @@ import postIdsStatuses from 'stores/PostIdsStatuses'
...
+    const { casts, merkleRoot } = await fetchThreadByPostId(postId)
+    const merkles = casts.map((cast) => cast.merkleRoot)
+
+    farcasterStore.threads[postId] = Promise.resolve(merkles)
     if (merkleRoot)
       postIdsStatuses.idToMerkleRoot[postId] = Promise.resolve(merkleRoot)
-    for (const cast of casts) {
+    casts.forEach((cast) => {
       farcasterStore.casts[cast.merkleRoot] = Promise.resolve(cast)
       if (cast.postId) {
```
Comment (T-Damer) [inline src/stores/FarcasterStore.ts:29]: Let's remove curlies

File: src/stores/PostStore.ts [hunk 7]
```diff
@@ -31,14 +33,15 @@ interface PostStoreType {
 
 const farcasterContract = getPostStorage()
 
-const limit = 20
+const limit = 5
```
Comment (T-Damer) [inline src/stores/PostStore.ts:36]: Let's have a little more? Maybe 10-15?

File: src/components/BlockchainList/ThreadPart.tsx (hunk 4 above)
Comment (smyzin) [reply src/components/BlockchainList/ThreadPart.tsx:96]: It's better to leave it as it is

File: src/stores/FarcasterStore.ts (hunk 6 above)
Comment (smyzin) [reply src/stores/FarcasterStore.ts:29]: The next function call takes three lines, and in order not to confuse the developer, I think it's better to leave the curly brace

File: src/stores/PostStore.ts (hunk 7 above)
Comment (smyzin) [reply src/stores/PostStore.ts:36]: Returned

**Review by T-Damer — APPROVED** [review]
NoiceNiceGIF

**Pull request merged**